icicle/include/icicle/backend/vec_ops_backend.h

```diff
@@ -53,6 +53,13 @@ namespace icicle {
   using scalarHighNonZeroIdxOpImpl = std::function<eIcicleError(
     const Device& device, const scalar_t* input, uint64_t size, const VecOpsConfig& config, int64_t* out_idx)>;
 
+  using programExecutionImpl = std::function<eIcicleError(
+    const Device& device,
+    std::vector<scalar_t*>& data,
```

Comment:
const

Reply:
data contains inputs and outputs therefore it cannot be const